task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -564,11 +565,27 @@ def run(ti: RuntimeTaskInstance, log: Logger) -> ToSupervisor | None:
                 msg = early_exit
                 return msg
 
-            result = _execute_task(context, ti)
+            # TODO: looks a bit forced, couldn't come up with a better way that won't require a lot of changes
```

Comment:
If you have any ideas to implement it in a better way considering the existing limitations, I'd be happy for suggestions